lib/email/receiver.rb

```diff
@@ -859,7 +859,8 @@ def process_destination(destination, user, body, elided)
 
         user ||= stage_from_user
 
-        if !user.has_trust_level?(SiteSetting.email_in_min_trust) && !sent_to_mailinglist_mirror?
+        if user.groups.any? && !user.in_any_groups?(SiteSetting.email_in_allowed_groups_map) &&
```

Comment:
Shouldn't need `user.groups.any?` here -- if they aren't in any groups then `in_any_groups?` should return false.

Reply:
So... ya we actually do need this. It's for "staged" users, so users emailing in, they don't have any groups, and so its different behavior than being an actual user that has a group (and trust level).

Reply:
@martin-brennan oh look, checkout the `has_trust_level?` helper method:

https://github.com/discourse/discourse/blob/594f4235ccda80b5230bef47ee2870f115162690/app/models/user.rb#L1301-L1304

It has a specific `|| staged?` catch, so maybe in a separate PR we can add a similar case to the `in_any_groups?` helper method so that we can get rid of the `user.groups.any?` check. I'd have to think on it a bit though.

Reply:
Yes interesting, I would like to see this in a separate PR :) So are we saying that if the user is staged this `in_any_groups?` check would return true always?